openpype/hosts/houdini/hooks/set_default_display_and_view.py

```diff
@@ -0,0 +1,64 @@
+from openpype.lib.applications import PreLaunchHook, LaunchTypes
+
+
+class SetDefaultDisplayView(PreLaunchHook):
+    """Set default view and default display for houdini via OpenColorIO.
+
+    Houdini's defaultDisplay and defaultView are set by
+    setting 'OCIO_ACTIVE_DISPLAYS' and 'OCIO_ACTIVE_VIEWS'
+    environment variables respectively.
+
+    More info: https://www.sidefx.com/docs/houdini/io/ocio.html#set-up
+    """
+
+    app_groups = {"houdini"}
+    launch_types = {LaunchTypes.local}
+
+    def execute(self):
+
+        OCIO = self.launch_context.env.get("OCIO")
+
+        # This is a cheap way to skip this hook if either global color
+        # management or houdini color management was disabled because the
+        # OCIO var would be set by the global OCIOEnvHook
+        if not OCIO:
+            return
+
+        houdini_color_settings = \
+            self.data["project_settings"]["houdini"]["imageio"]["workfile"]
+
+        if not houdini_color_settings["enabled"]:
+            self.log.info(
+                "Houdini workfile color management is disabled."
+            )
+            return
+
+        # This is a way to get values specified by admins if they already
+        # added 'OCIO_ACTIVE_DISPLAYS', 'OCIO_ACTIVE_VIEWS' manually
+        # using Ayon global env vars or Ayon app env vars
+        # or Ayon houdini tool
+        OCIO_ACTIVE_DISPLAYS = self.launch_context.env.get(
+            "OCIO_ACTIVE_DISPLAYS", ""
+        )
+        OCIO_ACTIVE_VIEWS = self.launch_context.env.get(
+            "OCIO_ACTIVE_VIEWS", ""
+        )
+
+        # default_display and default_view
+        default_display = houdini_color_settings["default_display"]
+        default_display = ":".join([default_display, OCIO_ACTIVE_DISPLAYS])
```

Comment:
This will either prefix or suffix with `:` if either `default_display` or `OCIO_ACTIVE_DISPLAYS` is undefined.

I think you'd want to join only entries that are non-empty, so that if both are not specified you wouldn't be left with `:`

You can do either:
`str.strip(":")` just so there's never any prefixing or trailing `:` or maybe nicer is to only join non-empty values

```python
":".join(key for key in [default_display, OCIO_ACTIVE_DISPLAYS] if key)
```

Reply:
Or you might just as well be better of only doing any JOIN if `default_view` and `default_display` are set at all, and otherwise not touch the values alltogether:

```python
if default_view:
    default_view = ":".join(key for key in [default_display, OCIO_ACTIVE_DISPLAYS] if key)
    # add log here
    # set env var here
```

Reply:
I was going to do something similar but I found out that OCIO ignores redundant ":" 
I also thought about the following code which won't log or set any env if user provides empty value for `default_display` 
as `OCIO_ACTIVE_DISPLAYS` was already set by OP env system before calling the hook. 
```python
if default_display:
    default_display = ":".join([default_display, OCIO_ACTIVE_DISPLAYS])
    self.log.info(
            "Setting OCIO_ACTIVE_DISPLAYS environment to: {}"
            .format(default_display)
        )
    self.launch_context.env["OCIO_ACTIVE_DISPLAYS"] = default_display
```

Reply:
I mean - it's also just to make sure it's clean when looking at it from an artist/dev perspective but if we don't care sure it's fine to have them prefixed. However, I'd personally be quite confused if I entered the value without `:` in settings that it'd suddenly get one on running the application - especially if the other key is empty. It wouldn't be my expected behavior, even if OCIO could deal with that.

Reply:
fair!